plugin/src/automation/lift/liftconfig.ml

```diff
@@ -318,11 +334,43 @@ let initialize_packed_constrs c env sigma =
   let a_typ, b_typ = c.typs in
   let l = c.l in
   let sigma, a_constrs =
-    map_constrs
-      (fun env constr sigma -> expand_eta env sigma constr)
-      env
-      a_typ
-      sigma
+    match l.orn.kind with
+    | Algebraic _ | CurryRecord | SwapConstruct _ ->
+       map_constrs
+         (fun env constr sigma -> expand_eta env sigma constr)
+         env
+         a_typ
+         sigma
+    | UnpackSigma ->
+       let env_a_typ = zoom_env zoom_lambda_term env a_typ in
```

Comment:
This needs massive cleaning

Reply:
Also an explanation: this isn't a constructor for an inductive type. Rather, we make a "constructor" that takes in the different parts of the packed sigma type and sticks them together. This turns out to be very, very useful for the algorithm, since we treat lifting this exactly like we'd treat lifting constructors of inductive types.